### ISSUE TITLE
'COM Interop is not supported on this platform.' when writing debug info on Linux

### ISSUE THREAD

**Issue by NextTurn** [dotnet/roslyn#43534]
**Version Used**: *Microsoft.CodeAnalysis.CSharp* 3.5.0

**Steps to Reproduce**:

I'd guess this is because the "debug type" is "full" by default but I couldn't find an option to change it.

```cs
using System;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.Emit;

class Program
{
    private static void Main()
    {
        CSharpCompilation compilation = CSharpCompilation.Create("test")
            .WithReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location))
            .WithOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));

        EmitResult result = compilation.Emit("test.dll", "test.pdb");
        Console.WriteLine(result.Success);
        foreach (Diagnostic diagnostic in result.Diagnostics)
        {
            Console.WriteLine(diagnostic);
        }
    }
}
```

**Expected Behavior**:

Something better.

**Actual Behavior**:

```
False
error CS0041: Unexpected error writing debug information -- 'COM Interop is not supported on this platform.'
```

**Comment by NMSAzulX** [comment]
Me too..

**Comment by alienwow** [comment]
me too

env:
Mac v10.15.4
SDK: v3.1.201

**Comment by tmat** [comment]
You need to pass EmitOptions to Emit with debugInformationFormat set to DebugInformationFormat.PortablePdb.

**Comment by NMSAzulX** [comment]
@tmat 
```C#
public static EmitResult Emit(this Compilation compilation, string outputPath, string? pdbPath = null, string? xmlDocPath = null, string? win32ResourcesPath = null, IEnumerable<ResourceDescription>? manifestResources = null, CancellationToken cancellationToken = default(CancellationToken))
        {
            if (compilation == null)
            {
                throw new ArgumentNullException("compilation");
            }
            using (Stream peStream = FileUtilities.CreateFileStreamChecked(File.Create, outputPath, "outputPath"))
            {
                using (Stream pdbStream = (pdbPath == null) ? null : FileUtilities.CreateFileStreamChecked(File.Create, pdbPath, "pdbPath"))
                {
                    using (Stream xmlDocumentationStream = (xmlDocPath == null) ? null : FileUtilities.CreateFileStreamChecked(File.Create, xmlDocPath, "xmlDocPath"))
                    {
                        using (Stream win32Resources = (win32ResourcesPath == null) ? null : FileUtilities.CreateFileStreamChecked(File.OpenRead, win32ResourcesPath, "win32ResourcesPath"))
                        {
                            return compilation.Emit(peStream, pdbStream, xmlDocumentationStream, win32Resources, manifestResources, new EmitOptions(metadataOnly: false, (DebugInformationFormat)0, pdbPath, null, 0, 0uL), cancellationToken);
                        }
                    }
                }
            }
        }
```
so  I need rewrite a new extension method?

**Comment by NMSAzulX** [comment]
And what's different in` DebugInformationFormat.Pdb`  、`DebugInformationFormat.PortablePdb` and `DebugInformationFormat.Embedded` ?

**Comment by NMSAzulX** [comment]
Does the OptimizationLevel.Release parameter of Compilation affect DebugInformationFormat?

**Comment by NextTurn** [comment]
Thanks @tmat. 'COM Interop is not supported' doesn't indicate the cause of the failure. Could you perform an argument validation and throw 'Full debug information is not supported' instead?
